Cargo.toml

```diff
@@ -14,6 +14,9 @@ serde = { version = "1.0", optional = true }
 parity-scale-codec-derive = { path = "derive", version = "1.0", default-features = false, optional = true }
 bitvec = { version = "0.14.0", default-features = false, features = ["alloc"], optional = true }
 byte-slice-cast = { version = "0.3.1", optional = true }
+generic-array = { version = "0.12", optional = true }
+vecarray = { version = "0.1", optional = true }
+typenum = { version = "1.10", optional = true }
```

Comment:
do we need also to add `default-features = false` ? same for byte-slice-cast actually, is this needed

Reply:
Fixed! `generic-array` and `typenum` does not need `std` feature.

Reply:
`byte-slice-cast` pulls in `std` by default and does not support `no_std`. So when we're building `bitvec` it just won't work for `no_std` environments.